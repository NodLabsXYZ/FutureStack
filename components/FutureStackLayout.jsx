import { useRouter } from 'next/router'

import {
  FutureStackFooter,
  FutureStackHeader, FutureStackNavigation
} from '.'

const dark = ['', 'login', 'register']

const FutureStackLayout = ({ user, children }) => {
  const router = useRouter();
  const rootPath = router.pathname.split('/')[1] || '';
  const isDark = !user && dark.includes(rootPath)

  const gradient = isDark ? {
    background: 'linear-gradient(#0C1322 10%, #5B5258 60%, #ECA33B 85%, #DB5224 100%)'
  } : {}

  return (
    <div className={`w-fit sm:w-full ${isDark ? 'text-white' : 'text-slate-900'}`} style={gradient}>
      <div className='container px-6 mx-auto pb-36'>
        <FutureStackHeader darkMode={isDark} user={user} />
        {user && <FutureStackNavigation />}
        {children}
      </div>
      <FutureStackFooter darkMode={isDark}/>
    </div>
  )

}

export default FutureStackLayout