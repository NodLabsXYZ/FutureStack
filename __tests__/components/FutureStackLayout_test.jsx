import renderer from 'react-test-renderer';
const { act } = renderer;

import FutureStackLayout from '../../components/FutureStackLayout'

describe("FutureStackLayout", () => {
  let component = null;

  describe("with user", () => {

    beforeEach(async () => {        
      await act(async () => {
        component = renderer.create(
          <FutureStackLayout
            provider={provider}
            contract={contract}
            onMint={onMint}
          />
        )
      })
    })
  
    it("should render the minting button", () => {
      const buttons = component.root.findAllByType(MintPriceAndButton)
      expect(buttons).toHaveLength(1)
    })

  })

})