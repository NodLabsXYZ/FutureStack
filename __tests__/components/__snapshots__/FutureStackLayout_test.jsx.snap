// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FutureStackLayout with user should render 1`] = `
<div
  className="w-fit sm:w-full text-slate-900"
  style={Object {}}
>
  <div
    className="container px-6 mx-auto pb-36"
  >
    <div
      className="leading-10 text-slate-900 py-6"
    >
      <div
        className="justify-between flex flex-row"
      >
        <h1
          className="text-lg"
        >
          <a
            className="flex cursor-pointer"
            onClick={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2718%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="FutureStack Logo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="FutureStack Logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2Fimages%2Fstack.png&w=48&q=75"
                  srcSet="/_next/image?url=%2Fimages%2Fstack.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Fstack.png&w=48&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
            <div
              className="ml-1"
            >
              <span
                className="font-light"
              >
                Future
              </span>
              <span
                className="font-semibold"
              >
                Stack
              </span>
            </div>
          </a>
        </h1>
        <div
          className="-mt-2"
        >
          <span
            className="cursor-pointer"
            onClick={[Function]}
          >
            Sign Out
          </span>
        </div>
      </div>
    </div>
    <div>
      FutureStackNavigation
    </div>
    <div
      id="children"
    >
      Children
    </div>
  </div>
  <footer
    className="mt-36"
  >
    <div
      className="text-slate-900 text-sm text-right px-12 pt-3 pb-9"
    >
      © 2022 Copyright  
      <a
        href="https://futurestack.xyz/"
      >
        Future
        <span
          className="font-semibold"
        >
          Stack
        </span>
      </a>
    </div>
  </footer>
</div>
`;
